olp-cpp-sdk-dataservice-read/src/generated/api/StreamApi.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "StreamApi.h"
+
+#include <map>
+
+#include <olp/core/client/CancellationContext.h>
+#include <olp/core/client/HttpResponse.h>
+#include <olp/core/client/OlpClient.h>
+#include <olp/core/logging/Log.h>
+#include <olp/dataservice/read/model/Data.h>
+// clang-format off
+#include "generated/parser/SubscribeResponseParser.h"
+#include <olp/core/generated/parser/JsonParser.h>
+#include "generated/serializer/ConsumerPropertiesSerializer.h"
+#include "generated/serializer/JsonSerializer.h"
+// clang-format on
+
+namespace {
+constexpr auto kLogTag = "read::StreamApi";
+}  // namespace
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+StreamApi::SubscribeApiResponse StreamApi::Subscribe(
+    const client::OlpClient& client, const std::string& layer_id,
+    const boost::optional<std::string>& subscription_id,
+    const boost::optional<std::string>& mode,
+    const boost::optional<std::string>& consumer_id,
+    const boost::optional<ConsumerProperties>& subscription_properties,
+    const client::CancellationContext& context, std::string& x_correlation_id) {
+  std::string metadata_uri = "/layers/" + layer_id + "/subscribe";
```

Comment:
```suggestion
  const std::string metadata_uri = "/layers/" + layer_id + "/subscribe";
```
Prefer immutable variables by default

Reply:
This and the next suggestions should be fixed in the next reviews if you don't mind